// Copyright 2025 Bytedance Ltd. and/or its affiliates
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// Protocol definition for rollout cache service
syntax = "proto3";

package specrl_fix;

// The request message containing the prompt hash and responses
message UpdateCacheRequest {
  // String type prompt hash
  uint64 prompt_hash = 1;

  // List of List of integers for responses
  // This is represented as repeated nested messages in protobuf
  TokenList prompt = 2;
  repeated TokenList responses = 3;
}

// A list of tokens (integers)
message TokenList {
  repeated int32 tokens = 1;
}

// The response message
message UpdateCacheResponse {
  bool success = 1;
}

// The rollout cache service definition
service RolloutCacheService {
  // RPC to update the cache with prompt hash and responses
  rpc UpdateCache(UpdateCacheRequest) returns (UpdateCacheResponse) {}
}
